Fix #47 by explicitly using Error.message to generate Notifications

florianb opened the pull request:
* fixes #47 
* fixes malformed tests
* ⚠️ this bug is untestable, since i wasn't able to open a TextEditor in a state which would provoke this Error.

sindresorhus commented:
Seems like unrelated commits got included here, like https://github.com/sindresorhus/atom-editorconfig/pull/84/commits/a8d2e467f7e4e66faf58dba112e48a38a14c0737.

sindresorhus commented on commands/generate.js:
```diff
@@ -46,7 +46,7 @@ trim_trailing_whitespace = false
 			fs.writeFile(configFile, ret, err => {
 				if (err) {
-					atom.notifications.addError(err);
+					atom.notifications.addError(err.message);
```
Can you add the `err.stack` as the `detail` option?

What do you think about opening an issue on Atom for `addError` and `addFatal` accepting an Error object? That would be a nice shorthand for displaying errors, and I bet most people forget to include the stack, like here.

florianb replied:
That's a good suggestion - i'll put it up.

florianb commented:
:rage: I guess i really have a problem with git-clients. This was my first (and last) PR with GitKraken.

sindresorhus commented on spec/iss67-spec.js:
```diff
@@ -10,7 +10,9 @@ import {init as generateConfig} from '../commands/generate';
-		waitsForPromise(() => atom.packages.activatePackage('editorconfig'));
+		waitsForPromise(() => {
+			return atom.packages.activatePackage('editorconfig');
+		});
```
This is exactly the same as the former, and the former is preferred.

florianb replied:
Really? i thought the return statement would be absolutely necessary. I'll rewind these kinds of lines.

sindresorhus replied:
Inline arrow functions implicitly returns. The previous and current has the exact same behavior.

florianb replied:
Thanks.. sorry for acting that naive.

sindresorhus commented:
If you're on OS X, I would strongly recommend http://gitup.co ;)

florianb commented:
@sindresorhus - i rewrote the history (unfortunately gitup [which is really good -- 💯 ] set you as the author for https://github.com/sindresorhus/atom-editorconfig/pull/84/commits/93adadb3606c5a5ee965569859d64a729c2222b8)) and added the stacktrace as detail.

sindresorhus commented on package.json:
```diff
@@ -1,6 +1,6 @@
 {
   "name": "editorconfig",
-  "version": "1.3.0",
+  "version": "1.4.0",
```
Don't bump the version in a PR. `apm` bumps it for me when doing a release ;)

sindresorhus commented:
@florianb I merged and changed the author to you => https://github.com/sindresorhus/atom-editorconfig/commit/23b22995c7f933340a575e8ff32de186bd91992b
